specs/design-black-box.md

```diff
@@ -0,0 +1,58 @@
+# GitOps Engine Design - Black Box
+
+## Summary
+
+During the elaboration of [White box](./design-white-box.md) option, it was discovered that some components are similar at a high-level but still have a lot of differences in 
+design and implementation. This is not surprising because code was developed by different teams and with a focus on different use-cases. Given that it would require a lot of
+effort to resolve such differences it is proposed contributing missing features of one project into an engine of another and use that engine in both projects.
+
+## Proposal
+
+It is proposed to use Argo CD application controller as the base for the GitOps engine and contribute a set of Flux features into it. There are two main reasons to try using Argo CD as
+a base:
+- Argo CD uses Application abstraction to represent the desired state and target Kubernetes cluster. This abstraction works for both Argo CD and Flux cases.
+- The Argo CD controller leverages Kubernetes watch APIs instead of pulling. This enables Argo CD features such as Health assessment, UI and could provide better performance to
+Flux as well.
+
+Following Flux features are missing in Argo CD:
+
+- Manifest generation using .flux.yaml files.
+- GPG commit signatures verification - an ability to verify the commit signature before pushing changes to the Kubernetes.
+- Namespace mode - an ability to control only given namespace in the target cluster. Currently, Argo CD requires read access in all namespaces.
+
+These features must be contributed to Argo CD before moving Argo CD controller into Argo Flux GitOps engine repo.
+
+Flux additionally provides the ability to monitor Docker registry and automatically push changes to the Git repository when a new image is released. Both teams feel the this should not
+be a part of GitOps engine. So it is proposed to keep the feature only in Flux for now and then work together to move it into a separate component that would work for both Flux
+and Argo CD.
+
+### Implementation Details/Notes/Constraints
+
+The proposed approach is based on the assumption that Argo CD engine is flexible enough to cover all Flux use-cases and can be easily integrated into Argo CD. This assumption might
+be wrong. So if we are missing some important Flux feature the approach might be no feasible.
+
+### Risks and Mitigations
+
+To mitigate the risk of let's start from POC to catch possible blockers earlier. 
```

Comment:
Missing risk :)

Reply:
Oops, looks like it was cut off during rewriting of other section. Fixed.